packages/babel-helpers/src/helpers/wrapRegExp.ts

```diff
@@ -72,9 +72,15 @@ export default function _wrapRegExp(this: any): RegExp {
       ).call(
         this,
         str,
-        substitution.replace(/\$<([^>]+)>/g, function (_, name) {
-          var group = groups[name];
-          return "$" + (Array.isArray(group) ? group.join("$") : group);
+        substitution.replace(/\$<([^>]+)(>|\$)/g, function (match, name, end) {
+          if (end === "$") {
```

Comment:
End will not actually be `$` but `""`, by `$` I meant the regexp marker for the end of the string :)

Reply:
Oh I see, somehow they both work because the quadratic pass is interrupted. I will resort to the line terminator approach as the footprint is smaller.